src/System.Security.Cryptography.X509Certificates/tests/CertificateCreation/CertificateRequestChainTests.cs

```diff
@@ -0,0 +1,313 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Xunit;
+
+namespace System.Security.Cryptography.X509Certificates.Tests.CertificateCreation
+{
+    public static class CertificateRequestChainTests
+    {
+        [Fact]
+        public static void CreateChain_ECC()
+        {
+            using (ECDsa rootKey = ECDsa.Create(ECCurve.NamedCurves.nistP521))
+            using (ECDsa intermed1Key = ECDsa.Create(ECCurve.NamedCurves.nistP384))
+            using (ECDsa intermed2Key = ECDsa.Create(ECCurve.NamedCurves.nistP384))
+            using (ECDsa leafKey = ECDsa.Create(ECCurve.NamedCurves.nistP256))
+            using (ECDsa leafPubKey = ECDsa.Create(leafKey.ExportParameters(false)))
+            {
+                CreateAndTestChain(
+                    rootKey,
+                    intermed1Key,
+                    intermed2Key,
+                    leafPubKey);
+            }
+        }
+
+        [Fact]
+        public static void CreateChain_RSA()
+        {
+            using (RSA rootKey = RSA.Create())
```

Comment:
I must have last tired this before rebasing where the new API existed.  It worked this time, so: done.